tests/AppBundle/EventListener/RedirectToPreferredLocaleListenerTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\AppBundle\EventListener;
+
+use AppBundle\EventListener\RedirectToPreferredLocaleListener;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Unit test for the event listener.
+ *
+ * See http://symfony.com/doc/current/book/testing.html#unit-tests
+ *
+ * Execute the application tests using this command (requires PHPUnit to be installed):
+ *
+ *     $ cd your-symfony-project/
+ *     $ ./vendor/bin/phpunit
+ *
+ * @author Michael COULLERET <michael.coulleret@gmail.com>
+ */
+class RedirectToPreferredLocaleListenerTest extends \PHPUnit_Framework_TestCase
+{
+    private $request;
+    private $urlGenerator;
+    private $event;
+    private $listener;
+
+    public function setUp()
+    {
+        $this->request = $this->prophesize(Request::class);
+        $this->urlGenerator = $this->prophesize(UrlGeneratorInterface::class);
+        $this->event = $this->prophesize(GetResponseEvent::class);
+        $this->listener = new RedirectToPreferredLocaleListener($this->urlGenerator->reveal(), 'en|fr|es');
+    }
+
+    public function testListenerOnKernelRequestUnexpectedValueException()
+    {
+        $this->setExpectedException(\UnexpectedValueException::class);
```

Comment:
Case 1 is not correct because an "explose" always returns an array.
http://php.net/manual/en/function.explode.php
In my opinion it is a logical problem in listening class. Cc @javiereguiluz